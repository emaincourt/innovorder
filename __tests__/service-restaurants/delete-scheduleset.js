import Boom from 'boom';

beforeEach(() => {
  jest.resetModules();
});

describe('Handler', () => {
  it('deletes a scheduleset according to its id', async () => {
    expect.assertions(4);
    jest.mock('../../sequelize/models', () => ({
      schedule: {
        findAll: (query) => {
          expect(query).toEqual({ where: { scheduleSetId: 'any-scheduleset-id' } });
          return Promise.resolve([{
            destroy: () => Promise.resolve(),
          }]);
        },
      },
      scheduleSet: {
        findById: (id) => {
          expect(id).toEqual('any-scheduleset-id');
          return Promise.resolve({
            destroy: () => Promise.resolve(),
          });
        },
      },
    }));
    const DeleteScheduleSet = require('../../services/service-restaurants/delete-scheduleset').default;
    await expect(
      DeleteScheduleSet.handler({
        params: {
          scheduleSetId: 'any-scheduleset-id',
        },
      }, (response) => {
        expect(response).toEqual({});
      }),
    ).resolves.toMatchSnapshot();
  });

  it('forwards the error if one gets thrown during the deletion', async () => {
    expect.assertions(4);
    jest.mock('../../sequelize/models', () => ({
      schedule: {
        findAll: (query) => {
          expect(query).toEqual({ where: { scheduleSetId: 'any-scheduleset-id' } });
          return Promise.resolve([{
            destroy: () => Promise.resolve(),
          }]);
        },
      },
      scheduleSet: {
        findById: (id) => {
          expect(id).toEqual('any-scheduleset-id');
          return Promise.resolve({
            destroy: () => Promise.reject(new Error()),
          });
        },
      },
    }));
    const DeleteScheduleSet = require('../../services/service-restaurants/delete-scheduleset').default;
    await expect(
      DeleteScheduleSet.handler({
        params: {
          scheduleSetId: 'any-scheduleset-id',
        },
      }, (response) => {
        expect(response).toEqual(Boom.wrap(new Error()));
      }),
    ).resolves.toMatchSnapshot();
  });
});
