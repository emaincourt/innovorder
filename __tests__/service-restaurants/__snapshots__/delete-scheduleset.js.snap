// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler deletes a scheduleset according to its id 1`] = `undefined`;

exports[`Handler forwards the error if one gets thrown during the deletion 1`] = `undefined`;
